Remove QGL from PyQLab.

blakejohnson opened the pull request:
QGL has moved to https://github.com/BBN-Q/QGL. The common JSON library manipulation code is now at https://github.com/BBN-Q/JSONLibraryUtils.

Still TODO before merging this:
- [x] Add JSONLibraryUtils submodule and remove that code from here
- [x] Write a migrator for the revised ChannelLibrary schema.
- [x] Update the settings validators
- [ ] User testing (we don't currently have automated GUI testing).

blakejohnson committed:
Remove QGL.

blakejohnson committed:
Remove QGL IR translators.

blakejohnson committed:
WIP: Re-wire ExpSetttingGUI -> ChannelLibraryView.

Need to test various options for channel connections and whether or not they get
wired appropriately.

blakejohnson committed:
Remove code that was moved to the translators.

blakejohnson committed:
Update Channel translator in ChannelsViews.

blakejohnson committed:
Remove startup.py... it is no longer needed.

blakejohnson committed:
Remove QGL tests.

blakejohnson committed:
Remove crufty requirements.txt

We can bring it back if we decide to start updating it.

blakejohnson committed:
Use JSONMigrators from JSONLibraryUtils.

blakejohnson committed:
Update settings validator for schema changes.

blakejohnson committed:
Replace FileWatcher and JSONUtils with JSONLibraryUtils versions.

blakejohnson commented:
Well, I should fix the broken unit tests, but otherwise I think this is ready to go modulo some user testing.

blakejohnson committed:
Bump version of JSONLibraryUtils.

caryan committed:
:fire: unused `Pattern` imports

caryan committed:
Merge pull request #108 from BBN-Q/fix/import

Remove unused pattern imports

caryan commented:
@blakejohnson do we need to ensure both QGL and PyQLab configs are pointing to the same channel library file?

blakejohnson committed:
Fix JSON unit tests.

1. Removed the dependency on APS2 helper.
2. Connected ExpSettingsVal to the updated test dictionaries.
3. Fixed validate_atom to handle sub-dictionaries.

blakejohnson committed:
Add QGL install to travis script.

blakejohnson committed:
Update README.
